src/lib/services/index.ts

```diff
@@ -1,32 +1,105 @@
-export * as registerForeignDevice from './register-foreign-device'
-export * as reinitializeDevice from './reinitialize-device'
-export * as subscribeCov from './subscribe-cov'
-export * as subscribeProperty from './subscribe-property'
-export * as timeSync from './time-sync'
-export * as timeSyncUTC from './time-sync'
-export * as whoHas from './who-has'
-export * as whoIs from './who-is'
-export * as writeProperty from './write-property'
-export * as writePropertyMultiple from './write-property-multiple'
+// Import abstract classes
+import { BacnetService, BacnetAckService } from './AbstractServices'
+
+// Import service classes
+import AddListElement from './AddListElement'
+import AlarmAcknowledge from './AlarmAcknowledge'
+import AlarmSummary from './AlarmSummary'
+import AtomicReadFile from './AtomicReadFile'
+import AtomicWriteFile from './AtomicWriteFile'
+import CovNotify from './CovNotify'
+import CreateObject from './CreateObject'
+import DeleteObject from './DeleteObject'
+import DeviceCommunicationControl from './DeviceCommunicationControl'
+import Error from './Error'
+import EventInformation from './EventInformation'
+import EventNotifyData from './EventNotifyData'
+import GetEnrollmentSummary from './GetEnrollmentSummary'
+import GetEventInformation from './GetEventInformation'
+import IAm from './IAm'
+import IHave from './IHave'
+import LifeSafetyOperation from './LifeSafetyOperation'
+import PrivateTransfer from './PrivateTransfer'
+import ReadProperty from './ReadProperty'
+import ReadPropertyMultiple from './ReadPropertyMultiple'
+import ReadRange from './ReadRange'
+import RegisterForeignDevice from './RegisterForeignDevice'
+import ReinitializeDevice from './ReinitializeDevice'
+import SubscribeCov from './SubscribeCov'
+import SubscribeProperty from './SubscribeProperty'
+import TimeSync from './TimeSync'
+import WhoHas from './WhoHas'
+import WhoIs from './WhoIs'
+import WriteProperty from './WriteProperty'
+import WritePropertyMultiple from './WritePropertyMultiple'
+
+// Export abstract classes
+export { BacnetService, BacnetAckService }
+
+// Export service classes
+export { AddListElement }
+export { AlarmAcknowledge }
+export { AlarmSummary }
+export { AtomicReadFile }
+export { AtomicWriteFile }
+export { CovNotify }
+export { CreateObject }
+export { DeleteObject }
+export { DeviceCommunicationControl }
+export { Error }
+export { EventInformation }
+export { EventNotifyData }
+export { GetEnrollmentSummary }
+export { GetEventInformation }
+export { IAm }
+export { IHave }
+export { LifeSafetyOperation }
+export { PrivateTransfer }
+export { ReadProperty }
+export { ReadPropertyMultiple }
+export { ReadRange }
+export { RegisterForeignDevice }
+export { ReinitializeDevice }
+export { SubscribeCov }
+export { SubscribeProperty }
+export { TimeSync }
+export { WhoHas }
+export { WhoIs }
+export { WriteProperty }
+export { WritePropertyMultiple }
+
+// Create and export ServicesMap
+export const ServicesMap = {
```

Comment:
I would explicitly add a type here.

Record<string, BacnetService | BacnetAckService>

Not sure if it is automatically inferred in client?